src/rhsmlib/facts/virt.py

```diff
@@ -148,6 +148,14 @@ def get_all(self):
         virt_what_info = virt_what_collector.get_all()
         virt_info.update(virt_what_info)
 
+        # Ensure we do not gather the virt.uuid fact for host_types we cannot
+        # See BZ#1438085
+        for no_uuid_host_type in VirtUuidCollector.no_uuid_platforms:
+            if virt_what_info.get('virt.host_type', None) is None:
+                continue
```

Comment:
I'm not sure I understand the logic here.  If `virt.host_type` is None, then the for loop will go through all values in `no_uuid_host_type` and then proceed to try and gather the virt uuid.  If we just want to stop immediately if it's none, let's use a `break` instead.  Also, is it appropriate to keep on and try to collect a virt uuid if the `virt.host_type` is None?